### ISSUE TITLE
verpi.service stuck at `loaded activating`

### ISSUE THREAD

**Issue by jpvalery** [gleich/verpi#1]
Hey man! 

Thanks for building this, it looks very cool. 

I wanted to give it a spin on a RPI3 but couldn't get the service to start. Even after multiple reboots, the service is still listed as `loaded activating`

Any idea what could be causing that?

Thanks in advance!

**Comment by gleich** [comment]
Hey there! Not super sure why that might be happening... Would mind doing the following for me?

1. Try running the binary manually (`/usr/local/bin/verpi` in your terminal). Does that work?
2. Send the top output (so without the logs) of command `systemctl status verpi.service`.
3. Try [uninstalling](https://github.com/gleich/verpi#-uninstalling-verpi), rebooting, and then [reinstalling](https://github.com/gleich/verpi#-installing-verpi).

Sorry to hear that your having issues!

**Comment by jpvalery** [comment]
Thanks for the fast reply! 

1. Here's the output: 
```
    INFO      | Sat Sep 25 22:39:37 UTC 2021 | Loading configuration
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23e860]

goroutine 1 [running]:
github.com/gleich/verpi/pkg/conf.Read()
	/home/pi/verpi_build/verpi/pkg/conf/read.go:40 +0x190
main.main()
	/home/pi/verpi_build/verpi/main.go:14 +0x14
```

2. Here's the output: 
```
● verpi.service - See the status of your vercel deployments on the pimoroni blinkt!
   Loaded: loaded (/etc/systemd/system/verpi.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Sat 2021-09-25 18:40:24 ED
  Process: 2488 ExecStart=/usr/local/bin/verpi (code=exited, status=2)
 Main PID: 2488 (code=exited, status=2)
```

3. I'll wait for your reply before uninstalling/reinstalling

Thanks again!

**Comment by gleich** [comment]
Looking into it now.

**Comment by gleich** [comment]
Ok, I think I fixed it. Just run the following command to rebuild the binary and get it all updated:

```txt
wget -q -O - https://raw.githubusercontent.com/gleich/verpi/master/setup.py | sudo python3 - install
```

**Comment by jpvalery** [comment]
Tried running that and rebooting but the service was still activating. 

Tried 1. above again and got a different message this time: 
```
pi@xxxx:~ $ /usr/local/bin/verpi
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Loading configuration
   SUCCESS    | Sat Sep 25 23:01:36 UTC 2021 | Loaded configuration
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Getting the user's vercel username
   SUCCESS    | Sat Sep 25 23:01:36 UTC 2021 | Got jpvalery's vercel username
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Setting up display
Press Control + C to stop
   SUCCESS    | Sat Sep 25 23:01:36 UTC 2021 | Setup display
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Loading configuration
   SUCCESS    | Sat Sep 25 23:01:36 UTC 2021 | Loaded configuration
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Getting deployments
   SUCCESS    | Sat Sep 25 23:01:36 UTC 2021 | Got data for 13 deployments
    INFO      | Sat Sep 25 23:01:36 UTC 2021 | Updating lights
panic: runtime error: index out of range [8] with length 8

goroutine 1 [running]:
github.com/alexellis/blinkt_go.(*Blinkt).SetPixel(...)
	/home/pi/verpi_build/goroot/pkg/mod/github.com/alexellis/blinkt_go@v0.0.0-20180120180744-cc0ca163e0bc/blinkt.go:136
github.com/gleich/verpi/pkg/lights.Update({0x1894180, {0x18a6090, 0x18}}, {0x14e6000, 0xd, 0x10}, 0x140a580)
	/home/pi/verpi_build/verpi/pkg/lights/update.go:18 +0x2a4
main.main()
	/home/pi/verpi_build/verpi/main.go:45 +0x30c
``` 

(For the record this is 100% not urgent—don't feel like you have to fix this on a Saturday evening :) )

**Comment by gleich** [comment]
don't worry, I'm just chillin right now. Imma go have dome dinner and I can fix it in a little bit.

**Comment by gleich** [comment]
@jpvalery should be all fixed :). Just update again and everything should be all set 🤞🏻

**Comment by jpvalery** [comment]
So I tried again running install and I had the same issue. So I ran the uninstaller, rebooted, ran the installer again, rebooted, and here's what I'm getting:
```
● verpi.service - See the status of your vercel deployments on the pimoroni blinkt!
   Loaded: loaded (/etc/systemd/system/verpi.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Sun 2021-09-26 12:20:07 ED
  Process: 1100 ExecStart=/usr/local/bin/verpi (code=exited, status=2)
 Main PID: 1100 (code=exited, status=2)
 ```

```
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Loading configuration
lumber.Logger{NormalOut:(*os.File)(0x209e118), ErrOut:(*os.File)(0x209e120), ExitCode:1, ShowStack:true, Timezone:(*time.Location)(0x452610), Padding:true, Multiline:false, ColoredOutput:true, TrueColor:true}
   SUCCESS    | Sun Sep 26 12:21:06 EDT 2021 | Loaded configuration
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Getting the user's vercel username
lumber.Logger{NormalOut:(*os.File)(0x209e118), ErrOut:(*os.File)(0x209e120), ExitCode:1, ShowStack:true, Timezone:(*time.Location)(0x452610), Padding:true, Multiline:false, ColoredOutput:true, TrueColor:true}
   SUCCESS    | Sun Sep 26 12:21:06 EDT 2021 | Got jpvalery's vercel username
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Setting up display
Press Control + C to stop
lumber.Logger{NormalOut:(*os.File)(0x209e118), ErrOut:(*os.File)(0x209e120), ExitCode:1, ShowStack:true, Timezone:(*time.Location)(0x452610), Padding:true, Multiline:false, ColoredOutput:true, TrueColor:true}
   SUCCESS    | Sun Sep 26 12:21:06 EDT 2021 | Setup display
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Loading configuration
lumber.Logger{NormalOut:(*os.File)(0x209e118), ErrOut:(*os.File)(0x209e120), ExitCode:1, ShowStack:true, Timezone:(*time.Location)(0x452610), Padding:true, Multiline:false, ColoredOutput:true, TrueColor:true}
   SUCCESS    | Sun Sep 26 12:21:06 EDT 2021 | Loaded configuration
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Getting deployments
lumber.Logger{NormalOut:(*os.File)(0x209e118), ErrOut:(*os.File)(0x209e120), ExitCode:1, ShowStack:true, Timezone:(*time.Location)(0x452610), Padding:true, Multiline:false, ColoredOutput:true, TrueColor:true}
   SUCCESS    | Sun Sep 26 12:21:06 EDT 2021 | Got data for 9 deployments
    INFO      | Sun Sep 26 12:21:06 EDT 2021 | Updating lights
panic: runtime error: index out of range [8] with length 8

goroutine 1 [running]:
github.com/alexellis/blinkt_go.(*Blinkt).SetPixel(...)
	/home/pi/verpi_build/goroot/pkg/mod/github.com/alexellis/blinkt_go@v0.0.0-20180120180744-cc0ca163e0bc/blinkt.go:136
github.com/gleich/verpi/pkg/lights.Update({0x209e118, 0x209e120, 0x1, 0x1, 0x452610, 0x1, 0x0, 0x1, 0x1}, {0x20a6350, ...}, ...)
	/home/pi/verpi_build/verpi/pkg/lights/update.go:18 +0x2bc
main.main()
	/home/pi/verpi_build/verpi/main.go:47 +0x3f4
```

Looks like it had an issue when getting deployments. Would you have an idea why?

**Comment by gleich** [comment]
Ah, I'm sorry! Had `<=` instead of `<` for the cap on how many LEDs should be updated. So it tries to update 9 LEDs even though there are only 8. Just pushed a fix 😅

**Comment by jpvalery** [comment]
![image](https://user-images.githubusercontent.com/4112343/134816462-9af2a68e-a4cd-4aa8-8071-571dc6626e81.jpeg)

![gif](https://media.giphy.com/media/xT4ApjjMoHdbCrVu0w/giphy.gif?cid=ecf05e47zqowf8z30nexsunndosjpjnp71v7067codvvjsr4&rid=giphy.gif&ct=g)

**Comment by jpvalery** [comment]
fixed in 76c6fff

**Comment by gleich** [comment]
yay!
